component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/util/TOTPUtil.java

```diff
@@ -149,14 +155,12 @@ public static String decrypt(String cipherText) throws CryptoException {
     public static String getTOTPIssuerDisplayName(String tenantDomain, AuthenticationContext context)
             throws TOTPException {
 
-        String issuer = null;
-        if (TOTPAuthenticatorConstants.SUPER_TENANT_DOMAIN.equals(tenantDomain) ||
-                Boolean.parseBoolean(getTOTPParameters().get(TOTPAuthenticatorConstants.TOTP_COMMON_ISSUER))) {
-            issuer = getTOTPParameters().get(TOTPAuthenticatorConstants.TOTP_ISSUER);
-        } else if (context == null) {
-            issuer = getIssuerFromRegistry(tenantDomain);
-        } else if (context.getProperty(TOTPAuthenticatorConstants.TOTP_ISSUER) != null) {
-            issuer = (String) context.getProperty(TOTPAuthenticatorConstants.TOTP_ISSUER);
+        String issuer = getIssuerFromParameters();
+        if (StringUtils.isBlank(issuer)) {
+            issuer = getIssuerFromBranding(tenantDomain);
+        }
+        if (StringUtils.isBlank(issuer)) {
+            issuer = getIssuerFromContextOrRegistry(tenantDomain, context);
         }
         if (StringUtils.isBlank(issuer)) {
```

Comment:
Hope this part is not invoked when authenticator app is enrolled from my account portal  myaccount as mentioned in https://github.com/wso2/product-is/issues/19830#issuecomment-1963722455
So, let's handle that case as well after a fallback mechansim of displayname resolving

Reply:
This is the same method used to get the issuer name in My Account too.